pkg/core/mpt/billet.go

```diff
@@ -230,19 +247,66 @@ func (b *Billet) traverse(curr Node, process func(node Node, nodeBytes []byte) b
 			}
 			n.Children[i] = r
 		}
-		return n, nil
+		return b.tryCollapseBranch(n), nil
```

Comment:
Why do we need this in `traverse`? I though we only traverse to server MPT data requests, so this MPT structure will be thrown away after request processing.

Reply:
`traverse` is also used during the cold start of Billet to decide which part of MPT are not yet restored, so we need this.

Imagine the situation when we're starting the server and almost the whole MPT is restored except for several nodes. We can't keep the whole restored MPT in memory, so we need to collapse those parts that are already restored.

Reply:
But you're right, we don't need this to reply to `GetMPTData` requests. But at the same time, it's a nice benefit for such request handling, because we don't occupy a lot of memory while traversing MPT and keep this structure as small as possible during response construction.